assets/js/util/i18n.js

```diff
@@ -19,22 +19,189 @@
+	}
+
+	if ( 10000 <= number ) {
+		return Math.round( number / 1000 );
+	}
+
+	if ( 1000 <= number ) {
+		return Math.round( number / 100 ) / 10;
+	}
+	return number;
+};
+
+/**
+ * Formats a large number for shortened display.
+ *
+ * @since 1.0.0
+ *
+ * @param {number} number The large number to format.
+ * @return {string} The formatted number.
+ */
+export const readableLargeNumber = ( number ) => {
+	const withSingleDecimal = {
+		minimumFractionDigits: 1,
+		maximumFractionDigits: 1,
+	};
+
+	// Numbers over 1,000,000 round normally and display a single decimal unless the decimal is 0.
+	if ( 1000000 <= number ) {
+		return sprintf(
+			// translators: %s: an abbreviated number in millions.
+			__( '%sM', 'google-site-kit' ),
+			numberFormat( prepareForReadableLargeNumber( number ), number % 10 === 0 ? {} : withSingleDecimal )
+		);
+	}
+
+	// Numbers between 10,000 and 1,000,000 round normally and have no decimals
+	if ( 10000 <= number ) {
+		return sprintf(
+			// translators: %s: an abbreviated number in thousands.
+			__( '%sK', 'google-site-kit' ),
+			numberFormat( prepareForReadableLargeNumber( number ) )
+		);
+	}
+
+	// Numbers between 1,000 and 10,000 round normally and display a single decimal unless the decimal is 0.
+	if ( 1000 <= number ) {
+		return sprintf(
+			// translators: %s: an abbreviated number in thousands.
+			__( '%sK', 'google-site-kit' ),
+			numberFormat( prepareForReadableLargeNumber( number ), number % 10 === 0 ? {} : withSingleDecimal )
+		);
+	}
+
+	return numberFormat( number );
+};
+
+/**
+ * Formats a number with unit using the JS Internationalization Number Format API.
+ *
+ * In addition to the supported 'style' values of the lower-level `numberFormat` function, this function
+ * supports two additional 'style' values 'metric' and 'seconds'.
+ *
+ * Another differentiation in behavior is that by default the function will use 'metric' formatting instead
+ * of 'decimal' formatting.
+ *
+ * @since n.e.x.t
+ *
+ * @param {number|string}                     number    The number to format.
+ * @param {(Intl.NumberFormatOptions|string)} [options] Formatting options or unit shorthand.
+ *                                                      Possible shorthand values are '%', 's',
+ *                                                      or a currency code.
+ * @return {string} The formatted number with unit.
+ */
+export const numFmt = ( number, options = {} ) => {
+	// Cast parsable values to numeric types.
+	number = isFinite( number ) ? number : Number( number );
+
+	if ( ! isFinite( number ) ) {
+		// eslint-disable-next-line no-console
+		console.warn( 'Invalid number', number, typeof number );
+		return null;
+	}
+
+	let formatOptions = {};
+
+	// Expand shorthand values for units.
+	if ( '%' === options ) {
+		formatOptions = {
+			style: 'percent',
+			maximumFractionDigits: 2,
+		};
+	} else if ( 's' === options ) {
+		formatOptions = {
+			style: 'seconds',
+		};
```

Comment:
I'm not sure this is really accurate. `s` for seconds is a unit, but I would expect it to be formatted as `X sec` or `Xs` (as formatted by `{ style: 'unit', unit: 'second' }`, but this is instead formatting as a human-readable _duration_ from the total given seconds by reducing the number in seconds to the equivalent number of larger units of time where applicable.

If we need to support a shorthand for this as a unit, I'm not sure there is an actual unit we can use to accurately represent this. Perhaps we should remove the concept of a "unit" from the second parameter (`percent` isn't actually a unit either).

That's all to say, I think it's okay to have a shorthand for this, but the actual (custom) `style` should probably be `duration` which would be easier to understand in its usage than `s` as well IMO.

Reply:
Using `duration` for the `style` sounds good to me.

Reply:
sounds good. About "unit" being ambiguous, shall we change `datapointUnit` and `changeUnit` to `datapointStyle` and `changeStyle`? Style can either be a custom definition or a unit 🤔

Reply:
@kostyalmm You updated the shorthand here to `duration`, but what we meant was to update the `style` to `duration`. The shorthand should be kept as `s` - which is more "unit"-like.